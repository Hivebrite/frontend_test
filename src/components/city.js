import React from 'react';
import styled from 'styled-components';

const Button = styled.button`
    width: 100%;
    height: 60px !important;
    border: 1px solid lightgrey;
    border-radius: 5px;
    text-align: center;
    margin-bottom: 10px;
    cursor: pointer;
    font-family: 'Helvetica';
    font-weight: bold;
    font-size: 14px;

    &:hover {
        background-color: lightblue;
        color: white;
    }
`;

const City = (props) => {

    return (
        <Button onClick={props.onChoice}>
            {props.name}
        </Button>
    )
};

export default City;

