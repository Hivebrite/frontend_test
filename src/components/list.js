import React, { useEffect, useState } from 'react';
import styled from 'styled-components'

import data from '../data/cities.json';
import City from './city';

const ListWrapper = styled.div`
    width: 20%;
    height: 90%;
    padding: 30px;
    border: 1px solid lightgrey;
    background-color: green;
    border-radius: 3px;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
`

const SearchInput = styled.input`
    width: 100%;
    padding: 5px;
    border: 1px solid lightgrey;
    border-radius: 3px;
    margin-bottom: 10px;
`
const List = () => {
    const [cities, setCities] = useState([]);
    const [filteredList, setFilteredList] = useState([]);
    const [search, setSearch] = useState('');

    const onFiltering = (e) => {
        setSearch(e.target.value);
    }

    useEffect(() => {
        setCities(data);
    }, []);

    useEffect(() => {
        if (cities && cities.length > 0) {
            console.log('CITIES:', cities.length);
            const temp = cities.slice(0, 10);
            console.log('list', temp);
            setFilteredList(temp);
        }
    }, [cities]);

    return (
        <>
            <ListWrapper>
                <SearchInput type="text" value={search} onChange={onFiltering} placeholder="Filter the list"/>
                {filteredList && filteredList.length > 0 && filteredList.map((element) => {
                    return <City name={element.city}/>
                })}
            </ListWrapper>
        </>
    )

};

export default List;

// const getData = () => {
//     fetch('./test.json'
//         , {
//             headers: {
//                 'Content-Type': 'application/json',
//                 'Accept': 'application/json'
//             }
//         }
//     )
//         .then((response) => {
//             console.log('response', response)
//             return response.json();
//         })
//         .then((myJson) => {
//             console.log('myJson', myJson);
//             setCities(myJson)
//         });
// }