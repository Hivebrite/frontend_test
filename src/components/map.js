import React, { useState, useEffect } from 'react';
import { MapContainer, TileLayer, Marker, Popup } from 'react-leaflet'
import styled from 'styled-components'

const MapWrapper = styled.div`
    width: calc(80% - 20px);
    height: 100%;
    margin-left: 20px;
    border: 1px solid lightgrey;
    border-radius: 3px;
`
const Map = (props) => {
    const [coordinates, setCoordinates] = useState([44.582076, 103.461760]);
    const [details, setDetails] = useState('');

    const DrawMap = () => {
        return (
            <MapContainer
                center={coordinates}
                zoom={13}
                scrollWheelZoom={true}
            >
                <TileLayer
                    attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
                    url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
                />
                <Marker position={coordinates}>
                    <Popup>{details}</Popup>
                </Marker>
            </MapContainer>
        )
    };

    useEffect(() => {
        setCoordinates([props.city.latitude, props.city.longitude]);
        setDetails(`
            state: ${props.city.state}
            population: ${props.city.population}
            growth 00-13: ${props.city.growth_from_2000_to_2013}
        `);
        DrawMap();
    }, [props.city]); //eslint-disable-line

    return (
        <>
            <MapWrapper>
                <DrawMap />
            </MapWrapper>
        </>
    )

};

export default Map;
