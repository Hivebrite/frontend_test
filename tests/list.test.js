import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import { ListContainer, ListElements, ListWrapper, SearchInput } from '../src/components/list';

configure({ adapter: new Adapter()});

const cities = [
    {
        "city": "New York",
        "growth_from_2000_to_2013": "4.8%",
        "latitude": 40.7127837,
        "longitude": -74.0059413,
        "population": "8405837",
        "rank": "1",
        "state": "New York"
    },
    {
        "city": "Los Angeles",
        "growth_from_2000_to_2013": "4.8%",
        "latitude": 34.0522342,
        "longitude": -118.2436849,
        "population": "3884307",
        "rank": "2",
        "state": "California"
    },
    {
        "city": "Chicago",
        "growth_from_2000_to_2013": "-6.1%",
        "latitude": 41.8781136,
        "longitude": -87.6297982,
        "population": "2718782",
        "rank": "3",
        "state": "Illinois"
    },
    {
        "city": "Houston",
        "growth_from_2000_to_2013": "11.0%",
        "latitude": 29.7604267,
        "longitude": -95.3698028,
        "population": "2195914",
        "rank": "4",
        "state": "Texas"
    },
    {
        "city": "Philadelphia",
        "growth_from_2000_to_2013": "2.6%",
        "latitude": 39.9525839,
        "longitude": -75.1652215,
        "population": "1553165",
        "rank": "5",
        "state": "Pennsylvania"
    },
    {
        "city": "Phoenix",
        "growth_from_2000_to_2013": "14.0%",
        "latitude": 33.4483771,
        "longitude": -112.0740373,
        "population": "1513367",
        "rank": "6",
        "state": "Arizona"
    },
    {
        "city": "San Antonio",
        "growth_from_2000_to_2013": "21.0%",
        "latitude": 29.4241219,
        "longitude": -98.49362819999999,
        "population": "1409019",
        "rank": "7",
        "state": "Texas"
    },
    {
        "city": "San Diego",
        "growth_from_2000_to_2013": "10.5%",
        "latitude": 32.715738,
        "longitude": -117.1610838,
        "population": "1355896",
        "rank": "8",
        "state": "California"
    },
    {
        "city": "Dallas",
        "growth_from_2000_to_2013": "5.6%",
        "latitude": 32.7766642,
        "longitude": -96.79698789999999,
        "population": "1257676",
        "rank": "9",
        "state": "Texas"
    },
    {
        "city": "San Jose",
        "growth_from_2000_to_2013": "10.5%",
        "latitude": 37.3382082,
        "longitude": -121.8863286,
        "population": "998537",
        "rank": "10",
        "state": "California"
    },
];

describe('Listing and filtering cities', () => {
    it('should have 4 styled components', () => {
        const container = renderer.create(<ListContainer />).toJSON;
        const elements = renderer.create(<ListElements />).toJSON;
        const wrapper = renderer.create(<ListWrapper />).toJSON;
        const searchInput = renderer.create(<SearchInput />).toJSON;
        
        expect(container).toMatchSnapshot();
        expect(elements).toMatchSnapshot();
        expect(wrapper).toMatchSnapshot();
        expect(searchIn).toMatchSnapshot();
        // const onFiltering = jest.fn();
        // const wrapper = shallow(<ListContainer />);

        // const input = wrapper.find(<SearchInput onChange={onFiltering} />)
        //     .simulate('change', { target: { value: 'new' } });
        // expect(wrapper.find(<ListContainer />)).toContain(<SearchInput />);
        // expect(onFiltering).toBeCalledWith('new');
        // expect(wrapper).toMatchSnapshot();
        // expect(wrapper.find(City)).toHaveLength(10);
    });
});
