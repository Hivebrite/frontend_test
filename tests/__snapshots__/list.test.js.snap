// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing and filtering cities should filter the list of cities 1`] = `[Function]`;

exports[`Listing and filtering cities should have 4 styled components 1`] = `[Function]`;

exports[`Listing and filtering cities should have 4 styled components 2`] = `[Function]`;

exports[`Listing and filtering cities should have 4 styled components 3`] = `[Function]`;

exports[`Listing and filtering cities should have 4 styled components 4`] = `[Function]`;
